substrate/client/transaction-pool/src/fork_aware_txpool/fork_aware_txpool.rs

```diff
@@ -469,12 +469,59 @@ where
 		self.view_store.futures_at(at)
 	}
 
+	/// Searches in the view store for a view by iterating through the fork of
+	/// the `at` block, up to the last finalized block.
+	///
+	/// Returns with a pair of a maybe view and a set of enacted blocks (possibly empty if no view
+	/// is found), usually starting at the last known block of the fork, up to the block
+	/// corresponding to the found view.
+	fn find_best_view(
+		&self,
+		at: &HashAndNumber<Block>,
```

Comment:
```suggestion
		at: &HashAndNumber<Block>,
		finalized: Block::Hash
```
or `upto_hash`.
Could also be number.

Reply:
Decided to pass a block number as a parameter. Change here: [9d47e15](https://github.com/paritytech/polkadot-sdk/pull/8533/commits/9d47e15417b4cf3fe54ec942fc883f852e494b32).